great_expectations/dataset/pandas_dataset.py

```diff
@@ -301,27 +301,31 @@ def expect_column_values_to_not_be_null(self, column,
         series = self[column]
         boolean_mapped_null_values = series.isnull()
 
-        element_count = int(len(series))
-        nonnull_values = series[boolean_mapped_null_values==False]
-        nonnull_count = int((boolean_mapped_null_values==False).sum())
+        element_count = len(series)
+        success_count = sum(~boolean_mapped_null_values)
+        unexpected_count = element_count - success_count
 
-        boolean_mapped_success_values = boolean_mapped_null_values==False
-        success_count = boolean_mapped_success_values.sum()
-
-        unexpected_list = [None for i in list(series[(boolean_mapped_success_values==False)])]
-        unexpected_index_list = list(series[(boolean_mapped_success_values==False)].index)
-        unexpected_count = len(unexpected_list)
+        unexpected_list = [None for i in range(unexpected_count)]
+        unexpected_index_list = boolean_mapped_null_values[boolean_mapped_null_values].index
 
         # Pass element_count instead of nonnull_count, because that's the right denominator for this expectation
         success, percent_success = self._calc_map_expectation_success(success_count, element_count, mostly)
 
-        return_obj = self._format_column_map_output(
-            result_format, success,
-            element_count, nonnull_count,
-            unexpected_list, unexpected_index_list
-        )
+        return {
+            "success": success,
+            "result": {
+                "element_count": element_count,
+                "missing_count": unexpected_count,
+                "missing_percent": float(unexpected_count) / element_count,
+                "unexpected_count": unexpected_count,
+                "unexpected_percent": float(unexpected_count) / element_count,
+                "partial_unexpected_list": unexpected_list[:min(5, len(unexpected_list))],
```

Comment:
Instead of hard-coding 5 as the length here, please convert to using the `parse_result_format()` function at line 298 and returning no more than `result_format['partial_unexpected_count']` elements.

Reply:
I'll take out the `min` part as well. I'm pretty sure pandas is able to handle index overflow very gracefully, but I'll also check just in case.